presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -371,9 +372,9 @@ public static QueryInfo createQueryInfo(
                 ImmutableSet.of(),
                 planAndMore.map(PlanAndMore::getPlan).map(Plan::getStatsAndCosts).orElseGet(StatsAndCosts::empty),
                 ImmutableList.of(),
-                ImmutableSet.of(),
-                ImmutableSet.of(),
-                ImmutableSet.of(),
+                planAndMore.map(PlanAndMore::getInvokedScalarFunctions).orElseGet(ImmutableSet::of),
```

Comment:
Let's add the `isLogInvokedFunctionNamesEnabled(getSession())` check to maintain consistent behavior with Presto

Reply:
I am adding this condition checks when PlanAndMore object is constructed, since the Presto code conditionally fetch those function information in stateMachine instead of QueryExecution.